{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "e3984b83",
   "metadata": {},
   "source": [
    "### Set up"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9e768ddf",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8a187f8e",
   "metadata": {},
   "source": [
    "### Load Bulk ATAC-seq file\n",
    "location: `~/CSE185FinalProject/Figure_1e/data/GSE132440_ATAC_PeakNorm.txt`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "38cb117c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peaks</th>\n",
       "      <th>D13_2_T1.Normal (C;RIK+PBS-d2)</th>\n",
       "      <th>D7_3_T1.Normal (C;RIK+PBS-d2)</th>\n",
       "      <th>D1_2_C-Ren-PBS-D2 -female-</th>\n",
       "      <th>D8_2_T2.Reg-ADM (C;RIK+Caer-d2)</th>\n",
       "      <th>D8_3_T2.Reg-ADM (C;RIK+Caer-d2)</th>\n",
       "      <th>D11_1_T2.Reg-ADM (C;RIK+Caer-d2)</th>\n",
       "      <th>D14_2_C-Ren-C-d2 (ADM-shRen+)</th>\n",
       "      <th>D6_1_C-Ren-C-d2 (ADM-shRen+)</th>\n",
       "      <th>D6_3_T3.KrasG12D (KC;RIK+PBS-d2)</th>\n",
       "      <th>...</th>\n",
       "      <th>D19_1_T4.Tum-ADR (KC;RIK+Caer-d2)</th>\n",
       "      <th>D19_3_T4.Tum-ADR (KC;RIK+Caer-d2)</th>\n",
       "      <th>D4_2_KC-REN-C-d2 (ADR-shRen+)</th>\n",
       "      <th>D11_3_KC-552OFF-C-d2  (ADR-sh552Off)</th>\n",
       "      <th>D2_1_T7.PDAC (KPflC-GEMMs)</th>\n",
       "      <th>D12_1_T7.PDAC (KPcr_organoids)</th>\n",
       "      <th>D12_2_T7.PDAC (KPcr_organoids)</th>\n",
       "      <th>D22_3_T7.PDAC (KPcr_organoids)</th>\n",
       "      <th>D3_2_T7.PDAC (KPR127H_shp53_orgs)</th>\n",
       "      <th>D3_3_T7.PDAC (KPR127H_Ren_orgs)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1_3001656_3001961</td>\n",
       "      <td>13.419960</td>\n",
       "      <td>5.367960</td>\n",
       "      <td>11.504871</td>\n",
       "      <td>2.645634</td>\n",
       "      <td>4.317972</td>\n",
       "      <td>6.110474</td>\n",
       "      <td>6.988018</td>\n",
       "      <td>4.690510</td>\n",
       "      <td>6.218671</td>\n",
       "      <td>...</td>\n",
       "      <td>5.297277</td>\n",
       "      <td>2.788978</td>\n",
       "      <td>7.399043</td>\n",
       "      <td>2.088005</td>\n",
       "      <td>12.564510</td>\n",
       "      <td>4.917989</td>\n",
       "      <td>3.185094</td>\n",
       "      <td>13.584760</td>\n",
       "      <td>6.75689</td>\n",
       "      <td>13.294766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1_3109719_3110326</td>\n",
       "      <td>9.585685</td>\n",
       "      <td>19.324655</td>\n",
       "      <td>34.514614</td>\n",
       "      <td>14.550989</td>\n",
       "      <td>12.474142</td>\n",
       "      <td>13.748566</td>\n",
       "      <td>29.564693</td>\n",
       "      <td>10.051092</td>\n",
       "      <td>9.567187</td>\n",
       "      <td>...</td>\n",
       "      <td>9.837801</td>\n",
       "      <td>7.669690</td>\n",
       "      <td>4.708482</td>\n",
       "      <td>4.698010</td>\n",
       "      <td>3.769353</td>\n",
       "      <td>6.762235</td>\n",
       "      <td>9.555281</td>\n",
       "      <td>81.508557</td>\n",
       "      <td>10.61797</td>\n",
       "      <td>5.908785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1_3192412_3193073</td>\n",
       "      <td>1.917137</td>\n",
       "      <td>8.588735</td>\n",
       "      <td>13.805846</td>\n",
       "      <td>8.598312</td>\n",
       "      <td>7.676395</td>\n",
       "      <td>6.110474</td>\n",
       "      <td>10.213258</td>\n",
       "      <td>12.061311</td>\n",
       "      <td>7.175390</td>\n",
       "      <td>...</td>\n",
       "      <td>6.810785</td>\n",
       "      <td>4.880712</td>\n",
       "      <td>6.053762</td>\n",
       "      <td>6.264014</td>\n",
       "      <td>6.282255</td>\n",
       "      <td>6.762235</td>\n",
       "      <td>11.678677</td>\n",
       "      <td>9.056506</td>\n",
       "      <td>14.47905</td>\n",
       "      <td>38.407103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1_3239108_3239709</td>\n",
       "      <td>15.337097</td>\n",
       "      <td>6.441552</td>\n",
       "      <td>32.213640</td>\n",
       "      <td>14.550989</td>\n",
       "      <td>14.393240</td>\n",
       "      <td>11.711741</td>\n",
       "      <td>10.213258</td>\n",
       "      <td>18.091966</td>\n",
       "      <td>15.785858</td>\n",
       "      <td>...</td>\n",
       "      <td>13.621570</td>\n",
       "      <td>16.733870</td>\n",
       "      <td>14.125445</td>\n",
       "      <td>9.396021</td>\n",
       "      <td>26.385471</td>\n",
       "      <td>11.065476</td>\n",
       "      <td>30.789239</td>\n",
       "      <td>31.697772</td>\n",
       "      <td>21.23594</td>\n",
       "      <td>29.543925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1_3351007_3351367</td>\n",
       "      <td>21.088508</td>\n",
       "      <td>7.515144</td>\n",
       "      <td>32.213640</td>\n",
       "      <td>8.598312</td>\n",
       "      <td>9.115719</td>\n",
       "      <td>7.638092</td>\n",
       "      <td>10.213258</td>\n",
       "      <td>10.051092</td>\n",
       "      <td>10.045546</td>\n",
       "      <td>...</td>\n",
       "      <td>11.351308</td>\n",
       "      <td>11.155913</td>\n",
       "      <td>10.089604</td>\n",
       "      <td>8.874020</td>\n",
       "      <td>15.077412</td>\n",
       "      <td>6.147486</td>\n",
       "      <td>8.493583</td>\n",
       "      <td>18.113013</td>\n",
       "      <td>33.78445</td>\n",
       "      <td>14.771963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103932</th>\n",
       "      <td>chrY_499480_499864</td>\n",
       "      <td>17.254234</td>\n",
       "      <td>9.662327</td>\n",
       "      <td>2.300974</td>\n",
       "      <td>5.291269</td>\n",
       "      <td>13.433691</td>\n",
       "      <td>14.766978</td>\n",
       "      <td>10.750798</td>\n",
       "      <td>16.751820</td>\n",
       "      <td>22.482889</td>\n",
       "      <td>...</td>\n",
       "      <td>33.297171</td>\n",
       "      <td>23.706315</td>\n",
       "      <td>37.667853</td>\n",
       "      <td>28.188062</td>\n",
       "      <td>28.898373</td>\n",
       "      <td>3.073743</td>\n",
       "      <td>6.370187</td>\n",
       "      <td>4.528253</td>\n",
       "      <td>0.96527</td>\n",
       "      <td>1.477196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103933</th>\n",
       "      <td>chrY_516124_516641</td>\n",
       "      <td>9.585685</td>\n",
       "      <td>8.588735</td>\n",
       "      <td>2.300974</td>\n",
       "      <td>5.952677</td>\n",
       "      <td>3.838197</td>\n",
       "      <td>7.128886</td>\n",
       "      <td>10.213258</td>\n",
       "      <td>2.010218</td>\n",
       "      <td>8.132109</td>\n",
       "      <td>...</td>\n",
       "      <td>22.702617</td>\n",
       "      <td>5.577957</td>\n",
       "      <td>32.286731</td>\n",
       "      <td>24.012053</td>\n",
       "      <td>17.590314</td>\n",
       "      <td>1.229497</td>\n",
       "      <td>3.185094</td>\n",
       "      <td>9.056506</td>\n",
       "      <td>0.96527</td>\n",
       "      <td>1.477196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103934</th>\n",
       "      <td>chrY_581156_582843</td>\n",
       "      <td>191.713707</td>\n",
       "      <td>122.389481</td>\n",
       "      <td>4.601949</td>\n",
       "      <td>142.202849</td>\n",
       "      <td>95.954936</td>\n",
       "      <td>106.424086</td>\n",
       "      <td>122.021553</td>\n",
       "      <td>137.364926</td>\n",
       "      <td>112.892803</td>\n",
       "      <td>...</td>\n",
       "      <td>116.540098</td>\n",
       "      <td>147.118603</td>\n",
       "      <td>113.676200</td>\n",
       "      <td>119.538263</td>\n",
       "      <td>98.003177</td>\n",
       "      <td>111.884252</td>\n",
       "      <td>32.912635</td>\n",
       "      <td>58.867291</td>\n",
       "      <td>0.96527</td>\n",
       "      <td>1.477196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103935</th>\n",
       "      <td>chrY_608935_609561</td>\n",
       "      <td>13.419960</td>\n",
       "      <td>6.441552</td>\n",
       "      <td>2.300974</td>\n",
       "      <td>8.598312</td>\n",
       "      <td>11.994367</td>\n",
       "      <td>10.184123</td>\n",
       "      <td>13.976037</td>\n",
       "      <td>8.710947</td>\n",
       "      <td>21.047811</td>\n",
       "      <td>...</td>\n",
       "      <td>17.405339</td>\n",
       "      <td>24.403560</td>\n",
       "      <td>22.869768</td>\n",
       "      <td>22.446049</td>\n",
       "      <td>25.129020</td>\n",
       "      <td>1.229497</td>\n",
       "      <td>2.123396</td>\n",
       "      <td>9.056506</td>\n",
       "      <td>1.93054</td>\n",
       "      <td>1.477196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103936</th>\n",
       "      <td>chrY_622075_623340</td>\n",
       "      <td>159.122377</td>\n",
       "      <td>142.787728</td>\n",
       "      <td>6.902923</td>\n",
       "      <td>141.541440</td>\n",
       "      <td>114.666148</td>\n",
       "      <td>120.681858</td>\n",
       "      <td>131.159731</td>\n",
       "      <td>163.497766</td>\n",
       "      <td>118.633115</td>\n",
       "      <td>...</td>\n",
       "      <td>124.107637</td>\n",
       "      <td>165.246962</td>\n",
       "      <td>138.563888</td>\n",
       "      <td>119.016262</td>\n",
       "      <td>96.746726</td>\n",
       "      <td>103.277771</td>\n",
       "      <td>28.665844</td>\n",
       "      <td>58.867291</td>\n",
       "      <td>0.96527</td>\n",
       "      <td>1.477196</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>103937 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       Peaks  D13_2_T1.Normal (C;RIK+PBS-d2)  \\\n",
       "0       chr1_3001656_3001961                       13.419960   \n",
       "1       chr1_3109719_3110326                        9.585685   \n",
       "2       chr1_3192412_3193073                        1.917137   \n",
       "3       chr1_3239108_3239709                       15.337097   \n",
       "4       chr1_3351007_3351367                       21.088508   \n",
       "...                      ...                             ...   \n",
       "103932    chrY_499480_499864                       17.254234   \n",
       "103933    chrY_516124_516641                        9.585685   \n",
       "103934    chrY_581156_582843                      191.713707   \n",
       "103935    chrY_608935_609561                       13.419960   \n",
       "103936    chrY_622075_623340                      159.122377   \n",
       "\n",
       "        D7_3_T1.Normal (C;RIK+PBS-d2)  D1_2_C-Ren-PBS-D2 -female-  \\\n",
       "0                            5.367960                   11.504871   \n",
       "1                           19.324655                   34.514614   \n",
       "2                            8.588735                   13.805846   \n",
       "3                            6.441552                   32.213640   \n",
       "4                            7.515144                   32.213640   \n",
       "...                               ...                         ...   \n",
       "103932                       9.662327                    2.300974   \n",
       "103933                       8.588735                    2.300974   \n",
       "103934                     122.389481                    4.601949   \n",
       "103935                       6.441552                    2.300974   \n",
       "103936                     142.787728                    6.902923   \n",
       "\n",
       "        D8_2_T2.Reg-ADM (C;RIK+Caer-d2)  D8_3_T2.Reg-ADM (C;RIK+Caer-d2)  \\\n",
       "0                              2.645634                         4.317972   \n",
       "1                             14.550989                        12.474142   \n",
       "2                              8.598312                         7.676395   \n",
       "3                             14.550989                        14.393240   \n",
       "4                              8.598312                         9.115719   \n",
       "...                                 ...                              ...   \n",
       "103932                         5.291269                        13.433691   \n",
       "103933                         5.952677                         3.838197   \n",
       "103934                       142.202849                        95.954936   \n",
       "103935                         8.598312                        11.994367   \n",
       "103936                       141.541440                       114.666148   \n",
       "\n",
       "        D11_1_T2.Reg-ADM (C;RIK+Caer-d2)  D14_2_C-Ren-C-d2 (ADM-shRen+)  \\\n",
       "0                               6.110474                       6.988018   \n",
       "1                              13.748566                      29.564693   \n",
       "2                               6.110474                      10.213258   \n",
       "3                              11.711741                      10.213258   \n",
       "4                               7.638092                      10.213258   \n",
       "...                                  ...                            ...   \n",
       "103932                         14.766978                      10.750798   \n",
       "103933                          7.128886                      10.213258   \n",
       "103934                        106.424086                     122.021553   \n",
       "103935                         10.184123                      13.976037   \n",
       "103936                        120.681858                     131.159731   \n",
       "\n",
       "        D6_1_C-Ren-C-d2 (ADM-shRen+)  D6_3_T3.KrasG12D (KC;RIK+PBS-d2)  ...  \\\n",
       "0                           4.690510                          6.218671  ...   \n",
       "1                          10.051092                          9.567187  ...   \n",
       "2                          12.061311                          7.175390  ...   \n",
       "3                          18.091966                         15.785858  ...   \n",
       "4                          10.051092                         10.045546  ...   \n",
       "...                              ...                               ...  ...   \n",
       "103932                     16.751820                         22.482889  ...   \n",
       "103933                      2.010218                          8.132109  ...   \n",
       "103934                    137.364926                        112.892803  ...   \n",
       "103935                      8.710947                         21.047811  ...   \n",
       "103936                    163.497766                        118.633115  ...   \n",
       "\n",
       "        D19_1_T4.Tum-ADR (KC;RIK+Caer-d2)  D19_3_T4.Tum-ADR (KC;RIK+Caer-d2)  \\\n",
       "0                                5.297277                           2.788978   \n",
       "1                                9.837801                           7.669690   \n",
       "2                                6.810785                           4.880712   \n",
       "3                               13.621570                          16.733870   \n",
       "4                               11.351308                          11.155913   \n",
       "...                                   ...                                ...   \n",
       "103932                          33.297171                          23.706315   \n",
       "103933                          22.702617                           5.577957   \n",
       "103934                         116.540098                         147.118603   \n",
       "103935                          17.405339                          24.403560   \n",
       "103936                         124.107637                         165.246962   \n",
       "\n",
       "        D4_2_KC-REN-C-d2 (ADR-shRen+)  D11_3_KC-552OFF-C-d2  (ADR-sh552Off)  \\\n",
       "0                            7.399043                              2.088005   \n",
       "1                            4.708482                              4.698010   \n",
       "2                            6.053762                              6.264014   \n",
       "3                           14.125445                              9.396021   \n",
       "4                           10.089604                              8.874020   \n",
       "...                               ...                                   ...   \n",
       "103932                      37.667853                             28.188062   \n",
       "103933                      32.286731                             24.012053   \n",
       "103934                     113.676200                            119.538263   \n",
       "103935                      22.869768                             22.446049   \n",
       "103936                     138.563888                            119.016262   \n",
       "\n",
       "        D2_1_T7.PDAC (KPflC-GEMMs)  D12_1_T7.PDAC (KPcr_organoids)  \\\n",
       "0                        12.564510                        4.917989   \n",
       "1                         3.769353                        6.762235   \n",
       "2                         6.282255                        6.762235   \n",
       "3                        26.385471                       11.065476   \n",
       "4                        15.077412                        6.147486   \n",
       "...                            ...                             ...   \n",
       "103932                   28.898373                        3.073743   \n",
       "103933                   17.590314                        1.229497   \n",
       "103934                   98.003177                      111.884252   \n",
       "103935                   25.129020                        1.229497   \n",
       "103936                   96.746726                      103.277771   \n",
       "\n",
       "        D12_2_T7.PDAC (KPcr_organoids)  D22_3_T7.PDAC (KPcr_organoids)  \\\n",
       "0                             3.185094                       13.584760   \n",
       "1                             9.555281                       81.508557   \n",
       "2                            11.678677                        9.056506   \n",
       "3                            30.789239                       31.697772   \n",
       "4                             8.493583                       18.113013   \n",
       "...                                ...                             ...   \n",
       "103932                        6.370187                        4.528253   \n",
       "103933                        3.185094                        9.056506   \n",
       "103934                       32.912635                       58.867291   \n",
       "103935                        2.123396                        9.056506   \n",
       "103936                       28.665844                       58.867291   \n",
       "\n",
       "        D3_2_T7.PDAC (KPR127H_shp53_orgs)  D3_3_T7.PDAC (KPR127H_Ren_orgs)  \n",
       "0                                 6.75689                        13.294766  \n",
       "1                                10.61797                         5.908785  \n",
       "2                                14.47905                        38.407103  \n",
       "3                                21.23594                        29.543925  \n",
       "4                                33.78445                        14.771963  \n",
       "...                                   ...                              ...  \n",
       "103932                            0.96527                         1.477196  \n",
       "103933                            0.96527                         1.477196  \n",
       "103934                            0.96527                         1.477196  \n",
       "103935                            1.93054                         1.477196  \n",
       "103936                            0.96527                         1.477196  \n",
       "\n",
       "[103937 rows x 24 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ATAC = pd.read_csv('~/CSE185FinalProject/Figure_1e/data/GSE132440_ATAC_PeakNorm.txt',\n",
    "                   sep=\"\\t\")\n",
    "ATAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "0de7fe3e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Peaks', 'Normal_ATAC2', 'Normal_ATAC3', 'Normal_ATAC1', 'Injury_ATAC4',\n",
       "       'Injury_ATAC5', 'Injury_ATAC1', 'Injury_ATAC2', 'Injury_ATAC3',\n",
       "       'Kras_ATAC2', 'Kras_ATAC3', 'Kras_ATAC1', 'KrasInjury_ATAC5',\n",
       "       'KrasInjury_ATAC2', 'KrasInjury_ATAC6', 'KrasInjury_ATAC3',\n",
       "       'KrasInjury_ATAC4', 'KrasInjury_ATAC1', 'PDAC_ATAC1', 'PDAC_ATAC2',\n",
       "       'PDAC_ATAC3', 'PDAC_ATAC4', 'PDAC_ATAC5', 'PDAC_ATAC6'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rename = {\n",
    "     \"D1_2_C-Ren-PBS-D2 -female-\": \"Normal_ATAC1\",\n",
    "     \"D13_2_T1.Normal (C;RIK+PBS-d2)\": \"Normal_ATAC2\",\n",
    "     \"D7_3_T1.Normal (C;RIK+PBS-d2)\": \"Normal_ATAC3\", \n",
    "     \"D11_1_T2.Reg-ADM (C;RIK+Caer-d2)\": \"Injury_ATAC1\",  \n",
    "     \"D14_2_C-Ren-C-d2 (ADM-shRen+)\": \"Injury_ATAC2\",  \n",
    "     \"D6_1_C-Ren-C-d2 (ADM-shRen+)\": \"Injury_ATAC3\",  \n",
    "     \"D8_2_T2.Reg-ADM (C;RIK+Caer-d2)\": \"Injury_ATAC4\",  \n",
    "     \"D8_3_T2.Reg-ADM (C;RIK+Caer-d2)\": \"Injury_ATAC5\",  \n",
    "     \"D13_3_T3.KrasG12D (KC;RIK+PBS-d2)\": \"Kras_ATAC1\",  \n",
    "     \"D6_3_T3.KrasG12D (KC;RIK+PBS-d2)\": \"Kras_ATAC2\",  \n",
    "     \"D8_1_T3.KrasG12D (KC;RIK+PBS-d2)\": \"Kras_ATAC3\", \n",
    "     \"D11_3_KC-552OFF-C-d2  (ADR-sh552Off)\": \"KrasInjury_ATAC1\",  \n",
    "     \"D18_3_T4.Tum-ADR (KC;RIK+Caer-d2)\": \"KrasInjury_ATAC2\",  \n",
    "     \"D19_3_T4.Tum-ADR (KC;RIK+Caer-d2)\": \"KrasInjury_ATAC3\",  \n",
    "     \"D4_2_KC-REN-C-d2 (ADR-shRen+)\": \"KrasInjury_ATAC4\",  \n",
    "     \"D7_1_T4.Tum-ADR (KC;RIK+Caer-d2)\": \"KrasInjury_ATAC5\",  \n",
    "     \"D19_1_T4.Tum-ADR (KC;RIK+Caer-d2)\": \"KrasInjury_ATAC6\",  \n",
    "     \"D2_1_T7.PDAC (KPflC-GEMMs)\": \"PDAC_ATAC1\",  \n",
    "     \"D12_1_T7.PDAC (KPcr_organoids)\": \"PDAC_ATAC2\",  \n",
    "     \"D12_2_T7.PDAC (KPcr_organoids)\": \"PDAC_ATAC3\", \n",
    "     \"D22_3_T7.PDAC (KPcr_organoids)\": \"PDAC_ATAC4\",\n",
    "     \"D3_2_T7.PDAC (KPR127H_shp53_orgs)\": \"PDAC_ATAC5\",\n",
    "     \"D3_3_T7.PDAC (KPR127H_Ren_orgs)\": \"PDAC_ATAC6\"\n",
    "}\n",
    "\n",
    "ATAC.rename(columns=rename, inplace=True)\n",
    "ATAC.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "174fe08d",
   "metadata": {},
   "source": [
    "### Load Bulk RNA-seq (Set 1)\n",
    "location: `data/Bulk_RNA-seq/GSE132326_Alonso-Curbelo_RNASeq_SET1.csv`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9a521a1d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Normal_RNA1</th>\n",
       "      <th>Normal_RNA2</th>\n",
       "      <th>Normal_RNA3</th>\n",
       "      <th>Normal_RNA4</th>\n",
       "      <th>RegADM_RNA1</th>\n",
       "      <th>RegADM_RNA2</th>\n",
       "      <th>RegADM_RNA3</th>\n",
       "      <th>RegADM_RNA4</th>\n",
       "      <th>RegADM_RNA5</th>\n",
       "      <th>...</th>\n",
       "      <th>PDAC_RNA1</th>\n",
       "      <th>PDAC_RNA2</th>\n",
       "      <th>PDAC_RNA3</th>\n",
       "      <th>Gene_name</th>\n",
       "      <th>chromosome_name</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "      <th>transcript_length</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Karyotype_band</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSMUSG00000000001</td>\n",
       "      <td>41</td>\n",
       "      <td>119</td>\n",
       "      <td>94</td>\n",
       "      <td>21</td>\n",
       "      <td>176</td>\n",
       "      <td>225</td>\n",
       "      <td>436</td>\n",
       "      <td>238</td>\n",
       "      <td>242</td>\n",
       "      <td>...</td>\n",
       "      <td>1532</td>\n",
       "      <td>1387</td>\n",
       "      <td>1357</td>\n",
       "      <td>Gnai3</td>\n",
       "      <td>3</td>\n",
       "      <td>108107280</td>\n",
       "      <td>108146146</td>\n",
       "      <td>3262</td>\n",
       "      <td>-</td>\n",
       "      <td>F2.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSMUSG00000000003</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Pbsn</td>\n",
       "      <td>X</td>\n",
       "      <td>77837901</td>\n",
       "      <td>77853623</td>\n",
       "      <td>902</td>\n",
       "      <td>-</td>\n",
       "      <td>B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSMUSG00000000028</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>29</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>168</td>\n",
       "      <td>53</td>\n",
       "      <td>111</td>\n",
       "      <td>Cdc45</td>\n",
       "      <td>16</td>\n",
       "      <td>18780447</td>\n",
       "      <td>18811987</td>\n",
       "      <td>2143</td>\n",
       "      <td>-</td>\n",
       "      <td>A3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSMUSG00000000031</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>207</td>\n",
       "      <td>48</td>\n",
       "      <td>142</td>\n",
       "      <td>48</td>\n",
       "      <td>263</td>\n",
       "      <td>...</td>\n",
       "      <td>420</td>\n",
       "      <td>255</td>\n",
       "      <td>296</td>\n",
       "      <td>H19</td>\n",
       "      <td>7</td>\n",
       "      <td>142575529</td>\n",
       "      <td>142578143</td>\n",
       "      <td>2286</td>\n",
       "      <td>-</td>\n",
       "      <td>F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSMUSG00000000037</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>Scml2</td>\n",
       "      <td>X</td>\n",
       "      <td>161117193</td>\n",
       "      <td>161258213</td>\n",
       "      <td>4847</td>\n",
       "      <td>+</td>\n",
       "      <td>F4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53460</th>\n",
       "      <td>ENSMUSG00000115846</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>AC107711.6</td>\n",
       "      <td>14</td>\n",
       "      <td>45541173</td>\n",
       "      <td>45543567</td>\n",
       "      <td>1088</td>\n",
       "      <td>-</td>\n",
       "      <td>C1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53461</th>\n",
       "      <td>ENSMUSG00000115847</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>AC109172.2</td>\n",
       "      <td>14</td>\n",
       "      <td>82855964</td>\n",
       "      <td>82856344</td>\n",
       "      <td>381</td>\n",
       "      <td>-</td>\n",
       "      <td>D3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53462</th>\n",
       "      <td>ENSMUSG00000115848</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>AC114008.2</td>\n",
       "      <td>15</td>\n",
       "      <td>37171058</td>\n",
       "      <td>37172404</td>\n",
       "      <td>1347</td>\n",
       "      <td>-</td>\n",
       "      <td>B3.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53463</th>\n",
       "      <td>ENSMUSG00000115849</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>AC156016.5</td>\n",
       "      <td>14</td>\n",
       "      <td>47261832</td>\n",
       "      <td>47262186</td>\n",
       "      <td>355</td>\n",
       "      <td>-</td>\n",
       "      <td>C1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53464</th>\n",
       "      <td>ENSMUSG00000115850</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>AC118639.2</td>\n",
       "      <td>15</td>\n",
       "      <td>25414637</td>\n",
       "      <td>25445640</td>\n",
       "      <td>223</td>\n",
       "      <td>-</td>\n",
       "      <td>B1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>53465 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       ID  Normal_RNA1  Normal_RNA2  Normal_RNA3  Normal_RNA4  \\\n",
       "0      ENSMUSG00000000001           41          119           94           21   \n",
       "1      ENSMUSG00000000003            0            0            0            0   \n",
       "2      ENSMUSG00000000028            2            2            1            7   \n",
       "3      ENSMUSG00000000031            0            3            5            5   \n",
       "4      ENSMUSG00000000037            0            0            0            0   \n",
       "...                   ...          ...          ...          ...          ...   \n",
       "53460  ENSMUSG00000115846            0            0            0            0   \n",
       "53461  ENSMUSG00000115847            0            0            0            0   \n",
       "53462  ENSMUSG00000115848            0            0            0            0   \n",
       "53463  ENSMUSG00000115849            0            0            0            0   \n",
       "53464  ENSMUSG00000115850            0            0            0            0   \n",
       "\n",
       "       RegADM_RNA1  RegADM_RNA2  RegADM_RNA3  RegADM_RNA4  RegADM_RNA5  ...  \\\n",
       "0              176          225          436          238          242  ...   \n",
       "1                0            0            0            0            0  ...   \n",
       "2                6            7           29           18           12  ...   \n",
       "3              207           48          142           48          263  ...   \n",
       "4                0            1            1            1            2  ...   \n",
       "...            ...          ...          ...          ...          ...  ...   \n",
       "53460            0            0            0            0            0  ...   \n",
       "53461            0            0            0            0            0  ...   \n",
       "53462            0            0            0            0            0  ...   \n",
       "53463            0            0            0            0            0  ...   \n",
       "53464            0            0            0            0            0  ...   \n",
       "\n",
       "       PDAC_RNA1  PDAC_RNA2  PDAC_RNA3   Gene_name  chromosome_name  \\\n",
       "0           1532       1387       1357       Gnai3                3   \n",
       "1              0          0          0        Pbsn                X   \n",
       "2            168         53        111       Cdc45               16   \n",
       "3            420        255        296         H19                7   \n",
       "4              8          1          4       Scml2                X   \n",
       "...          ...        ...        ...         ...              ...   \n",
       "53460          0          0          0  AC107711.6               14   \n",
       "53461          0          0          0  AC109172.2               14   \n",
       "53462          1          0          0  AC114008.2               15   \n",
       "53463          4          0          0  AC156016.5               14   \n",
       "53464          0          0          0  AC118639.2               15   \n",
       "\n",
       "       start_position  end_position  transcript_length  Strand  Karyotype_band  \n",
       "0           108107280     108146146               3262       -            F2.3  \n",
       "1            77837901      77853623                902       -               B  \n",
       "2            18780447      18811987               2143       -              A3  \n",
       "3           142575529     142578143               2286       -              F5  \n",
       "4           161117193     161258213               4847       +              F4  \n",
       "...               ...           ...                ...     ...             ...  \n",
       "53460        45541173      45543567               1088       -              C1  \n",
       "53461        82855964      82856344                381       -              D3  \n",
       "53462        37171058      37172404               1347       -            B3.1  \n",
       "53463        47261832      47262186                355       -              C1  \n",
       "53464        25414637      25445640                223       -              B1  \n",
       "\n",
       "[53465 rows x 27 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RNA = pd.read_csv('data/Bulk_RNA-seq/GSE132326_Alonso-Curbelo_RNASeq_SET1.csv')\n",
    "RNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "0362683e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>chromosome_name</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSMUSG00000000001</td>\n",
       "      <td>3</td>\n",
       "      <td>108107280</td>\n",
       "      <td>108146146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSMUSG00000000003</td>\n",
       "      <td>X</td>\n",
       "      <td>77837901</td>\n",
       "      <td>77853623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSMUSG00000000028</td>\n",
       "      <td>16</td>\n",
       "      <td>18780447</td>\n",
       "      <td>18811987</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSMUSG00000000031</td>\n",
       "      <td>7</td>\n",
       "      <td>142575529</td>\n",
       "      <td>142578143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSMUSG00000000037</td>\n",
       "      <td>X</td>\n",
       "      <td>161117193</td>\n",
       "      <td>161258213</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   ID chromosome_name  start_position  end_position\n",
       "0  ENSMUSG00000000001               3       108107280     108146146\n",
       "1  ENSMUSG00000000003               X        77837901      77853623\n",
       "2  ENSMUSG00000000028              16        18780447      18811987\n",
       "3  ENSMUSG00000000031               7       142575529     142578143\n",
       "4  ENSMUSG00000000037               X       161117193     161258213"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RNA_region = RNA[[\"ID\",\"chromosome_name\",\n",
    "                  \"start_position\",\"end_position\"]]\n",
    "RNA_region.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5b298e17",
   "metadata": {},
   "source": [
    "### Due to time, instead of doing the integrative analysis, just use the extended table 5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "ca62e55b",
   "metadata": {},
   "outputs": [],
   "source": [
    "integrative = pd.read_excel(\"data/41586_2020_3147_MOESM7_ESM.xlsx\",\n",
    "                            sheet_name=\"KrasG12DInjury_vs_Normal\", index_col=0)\n",
    "integrative.dropna(inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "165ab308",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>lfcSE</th>\n",
       "      <th>stat</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "      <th>ATAC-RNA_status</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Prss22</th>\n",
       "      <td>ENSMUSG00000045027</td>\n",
       "      <td>175.999100</td>\n",
       "      <td>9.181689</td>\n",
       "      <td>1.186109</td>\n",
       "      <td>7.741015</td>\n",
       "      <td>9.862618e-15</td>\n",
       "      <td>3.491408e-13</td>\n",
       "      <td>ATAC-GAIN_UP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chil4</th>\n",
       "      <td>ENSMUSG00000063779</td>\n",
       "      <td>187.038667</td>\n",
       "      <td>8.954918</td>\n",
       "      <td>1.758181</td>\n",
       "      <td>5.093286</td>\n",
       "      <td>3.519095e-07</td>\n",
       "      <td>3.742030e-06</td>\n",
       "      <td>ATAC-GAIN_UP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Anxa10</th>\n",
       "      <td>ENSMUSG00000031635</td>\n",
       "      <td>245.590307</td>\n",
       "      <td>8.709109</td>\n",
       "      <td>1.081537</td>\n",
       "      <td>8.052531</td>\n",
       "      <td>8.109935e-16</td>\n",
       "      <td>3.381342e-14</td>\n",
       "      <td>ATAC-GAIN_UP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Areg</th>\n",
       "      <td>ENSMUSG00000029378</td>\n",
       "      <td>201.135591</td>\n",
       "      <td>8.656658</td>\n",
       "      <td>1.178385</td>\n",
       "      <td>7.346204</td>\n",
       "      <td>2.039145e-13</td>\n",
       "      <td>5.885984e-12</td>\n",
       "      <td>ATAC-GAIN_UP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Areg</th>\n",
       "      <td>ENSMUSG00000029378</td>\n",
       "      <td>201.135591</td>\n",
       "      <td>8.656658</td>\n",
       "      <td>1.178385</td>\n",
       "      <td>7.346204</td>\n",
       "      <td>2.039145e-13</td>\n",
       "      <td>5.885984e-12</td>\n",
       "      <td>ATAC-GAIN_UP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           ID    baseMean  log2FoldChange     lfcSE      stat  \\\n",
       "Gene_name                                                                       \n",
       "Prss22     ENSMUSG00000045027  175.999100        9.181689  1.186109  7.741015   \n",
       "Chil4      ENSMUSG00000063779  187.038667        8.954918  1.758181  5.093286   \n",
       "Anxa10     ENSMUSG00000031635  245.590307        8.709109  1.081537  8.052531   \n",
       "Areg       ENSMUSG00000029378  201.135591        8.656658  1.178385  7.346204   \n",
       "Areg       ENSMUSG00000029378  201.135591        8.656658  1.178385  7.346204   \n",
       "\n",
       "                 pvalue          padj ATAC-RNA_status  \n",
       "Gene_name                                              \n",
       "Prss22     9.862618e-15  3.491408e-13    ATAC-GAIN_UP  \n",
       "Chil4      3.519095e-07  3.742030e-06    ATAC-GAIN_UP  \n",
       "Anxa10     8.109935e-16  3.381342e-14    ATAC-GAIN_UP  \n",
       "Areg       2.039145e-13  5.885984e-12    ATAC-GAIN_UP  \n",
       "Areg       2.039145e-13  5.885984e-12    ATAC-GAIN_UP  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "integrative.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "15323124",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ID                 False\n",
       "baseMean           False\n",
       "log2FoldChange     False\n",
       "lfcSE              False\n",
       "stat               False\n",
       "pvalue             False\n",
       "padj               False\n",
       "ATAC-RNA_status    False\n",
       "dtype: bool"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "integrative.isnull().any()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "a67057df",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>lfcSE</th>\n",
       "      <th>stat</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "      <th>ATAC-RNA_status</th>\n",
       "      <th>chromosome_name</th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSMUSG00000045027</td>\n",
       "      <td>175.999100</td>\n",
       "      <td>9.181689</td>\n",
       "      <td>1.186109</td>\n",
       "      <td>7.741015</td>\n",
       "      <td>9.862618e-15</td>\n",
       "      <td>3.491408e-13</td>\n",
       "      <td>ATAC-GAIN_UP</td>\n",
       "      <td>17</td>\n",
       "      <td>23993535</td>\n",
       "      <td>23998100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSMUSG00000063779</td>\n",
       "      <td>187.038667</td>\n",
       "      <td>8.954918</td>\n",
       "      <td>1.758181</td>\n",
       "      <td>5.093286</td>\n",
       "      <td>3.519095e-07</td>\n",
       "      <td>3.742030e-06</td>\n",
       "      <td>ATAC-GAIN_UP</td>\n",
       "      <td>3</td>\n",
       "      <td>106201490</td>\n",
       "      <td>106219507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSMUSG00000031635</td>\n",
       "      <td>245.590307</td>\n",
       "      <td>8.709109</td>\n",
       "      <td>1.081537</td>\n",
       "      <td>8.052531</td>\n",
       "      <td>8.109935e-16</td>\n",
       "      <td>3.381342e-14</td>\n",
       "      <td>ATAC-GAIN_UP</td>\n",
       "      <td>8</td>\n",
       "      <td>62057042</td>\n",
       "      <td>62123150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSMUSG00000029378</td>\n",
       "      <td>201.135591</td>\n",
       "      <td>8.656658</td>\n",
       "      <td>1.178385</td>\n",
       "      <td>7.346204</td>\n",
       "      <td>2.039145e-13</td>\n",
       "      <td>5.885984e-12</td>\n",
       "      <td>ATAC-GAIN_UP</td>\n",
       "      <td>5</td>\n",
       "      <td>91139599</td>\n",
       "      <td>91148432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSMUSG00000029378</td>\n",
       "      <td>201.135591</td>\n",
       "      <td>8.656658</td>\n",
       "      <td>1.178385</td>\n",
       "      <td>7.346204</td>\n",
       "      <td>2.039145e-13</td>\n",
       "      <td>5.885984e-12</td>\n",
       "      <td>ATAC-GAIN_UP</td>\n",
       "      <td>5</td>\n",
       "      <td>91139599</td>\n",
       "      <td>91148432</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   ID    baseMean  log2FoldChange     lfcSE      stat  \\\n",
       "0  ENSMUSG00000045027  175.999100        9.181689  1.186109  7.741015   \n",
       "1  ENSMUSG00000063779  187.038667        8.954918  1.758181  5.093286   \n",
       "2  ENSMUSG00000031635  245.590307        8.709109  1.081537  8.052531   \n",
       "3  ENSMUSG00000029378  201.135591        8.656658  1.178385  7.346204   \n",
       "4  ENSMUSG00000029378  201.135591        8.656658  1.178385  7.346204   \n",
       "\n",
       "         pvalue          padj ATAC-RNA_status chromosome_name  start_position  \\\n",
       "0  9.862618e-15  3.491408e-13    ATAC-GAIN_UP              17        23993535   \n",
       "1  3.519095e-07  3.742030e-06    ATAC-GAIN_UP               3       106201490   \n",
       "2  8.109935e-16  3.381342e-14    ATAC-GAIN_UP               8        62057042   \n",
       "3  2.039145e-13  5.885984e-12    ATAC-GAIN_UP               5        91139599   \n",
       "4  2.039145e-13  5.885984e-12    ATAC-GAIN_UP               5        91139599   \n",
       "\n",
       "   end_position  \n",
       "0      23998100  \n",
       "1     106219507  \n",
       "2      62123150  \n",
       "3      91148432  \n",
       "4      91148432  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "integrative_region = pd.merge(left=integrative, right=RNA_region, how='left',\n",
    "                              left_on='ID', right_on='ID')\n",
    "integrative_region.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "70bf53f2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ID                 False\n",
       "baseMean           False\n",
       "log2FoldChange     False\n",
       "lfcSE              False\n",
       "stat               False\n",
       "pvalue             False\n",
       "padj               False\n",
       "ATAC-RNA_status    False\n",
       "chromosome_name    False\n",
       "start_position     False\n",
       "end_position       False\n",
       "dtype: bool"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "integrative_region.isnull().any()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "4bc50102",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrom</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr17</th>\n",
       "      <td>23993535</td>\n",
       "      <td>23998100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr3</th>\n",
       "      <td>106201490</td>\n",
       "      <td>106219507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr8</th>\n",
       "      <td>62057042</td>\n",
       "      <td>62123150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr5</th>\n",
       "      <td>91139599</td>\n",
       "      <td>91148432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr5</th>\n",
       "      <td>91139599</td>\n",
       "      <td>91148432</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       start_position  end_position\n",
       "chrom                              \n",
       "chr17        23993535      23998100\n",
       "chr3        106201490     106219507\n",
       "chr8         62057042      62123150\n",
       "chr5         91139599      91148432\n",
       "chr5         91139599      91148432"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "integrative_region_bed = {\"chrom\": [],\n",
    "                          \"start_position\": [], \n",
    "                          \"end_position\": []}\n",
    "\n",
    "for chrom, start, end in zip(integrative_region.chromosome_name,\n",
    "                             integrative_region.start_position,\n",
    "                             integrative_region.end_position):\n",
    "    \n",
    "    integrative_region_bed[\"chrom\"].append(\"chr\" + str(chrom))\n",
    "    integrative_region_bed[\"start_position\"].append(start)\n",
    "    integrative_region_bed[\"end_position\"].append(end)\n",
    "\n",
    "integrative_region_BED_df = pd.DataFrame(integrative_region_bed)\n",
    "integrative_region_BED_df = integrative_region_BED_df.set_index(\"chrom\")\n",
    "integrative_region_BED_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "56b4bd39",
   "metadata": {},
   "outputs": [],
   "source": [
    "# create a pseudo bed file based on integrate information\n",
    "integrative_region_BED_df.to_csv('data/integrative.bed', sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "35326fe5",
   "metadata": {},
   "source": [
    "### The pseudo bed for scATACseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "27c6295f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import scanpy as scanpy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "f34948bf",
   "metadata": {},
   "outputs": [],
   "source": [
    "adata = scanpy.read_10x_h5(\"data/scATAC-seq/GSE137069_Merged_filtered_peak_bc_matrix.h5\",\n",
    "                           genome=None, gex_only=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "3a4e5437",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_ids</th>\n",
       "      <th>feature_types</th>\n",
       "      <th>genome</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1:3493532-3494115</th>\n",
       "      <td>chr1:3493532-3494115</td>\n",
       "      <td>Peaks</td>\n",
       "      <td>mm10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:3514498-3515608</th>\n",
       "      <td>chr1:3514498-3515608</td>\n",
       "      <td>Peaks</td>\n",
       "      <td>mm10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:3534683-3535113</th>\n",
       "      <td>chr1:3534683-3535113</td>\n",
       "      <td>Peaks</td>\n",
       "      <td>mm10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:3535378-3535755</th>\n",
       "      <td>chr1:3535378-3535755</td>\n",
       "      <td>Peaks</td>\n",
       "      <td>mm10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:3551128-3551665</th>\n",
       "      <td>chr1:3551128-3551665</td>\n",
       "      <td>Peaks</td>\n",
       "      <td>mm10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  gene_ids feature_types genome\n",
       "chr1:3493532-3494115  chr1:3493532-3494115         Peaks   mm10\n",
       "chr1:3514498-3515608  chr1:3514498-3515608         Peaks   mm10\n",
       "chr1:3534683-3535113  chr1:3534683-3535113         Peaks   mm10\n",
       "chr1:3535378-3535755  chr1:3535378-3535755         Peaks   mm10\n",
       "chr1:3551128-3551665  chr1:3551128-3551665         Peaks   mm10"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "scATAC = pd.DataFrame(adata.var)\n",
    "scATAC.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "7fdc0a55",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start_position</th>\n",
       "      <th>end_position</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrom</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1</th>\n",
       "      <td>3493532</td>\n",
       "      <td>3494115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1</th>\n",
       "      <td>3514498</td>\n",
       "      <td>3515608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1</th>\n",
       "      <td>3534683</td>\n",
       "      <td>3535113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1</th>\n",
       "      <td>3535378</td>\n",
       "      <td>3535755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1</th>\n",
       "      <td>3551128</td>\n",
       "      <td>3551665</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      start_position end_position\n",
       "chrom                            \n",
       "chr1         3493532      3494115\n",
       "chr1         3514498      3515608\n",
       "chr1         3534683      3535113\n",
       "chr1         3535378      3535755\n",
       "chr1         3551128      3551665"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "scATAC_bed = {\"chrom\": [],\n",
    "              \"start_position\":[],\n",
    "              \"end_position\":[]}\n",
    "\n",
    "for gene_ids in scATAC.gene_ids:\n",
    "    chrom_region = gene_ids.split(\":\")\n",
    "    chrom = chrom_region[0]\n",
    "    region = chrom_region[1]\n",
    "    \n",
    "    start_end = region.split(\"-\")\n",
    "    start = start_end[0]\n",
    "    end = start_end[1]\n",
    "    \n",
    "    scATAC_bed[\"chrom\"].append(chrom)\n",
    "    scATAC_bed[\"start_position\"].append(start)\n",
    "    scATAC_bed[\"end_position\"].append(end)\n",
    "    \n",
    "scATAC_BED_df = pd.DataFrame(scATAC_bed)\n",
    "scATAC_BED_df = scATAC_BED_df.set_index(\"chrom\")\n",
    "scATAC_BED_df.head()    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "0cbb17ff",
   "metadata": {},
   "outputs": [],
   "source": [
    "# create a pseudo bed file based on scATACseq information\n",
    "scATAC_BED_df.to_csv('data/scATAC.bed', sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d42f11e4",
   "metadata": {},
   "source": [
    "## Find motif\n",
    "- AP-1: AP-1(bZIP)/ThioMac-PU.1-ChIP-Seq [GSM537983](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM537983)\n",
    "- NR5A2: [GSM846875](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM846875)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8e5ff416",
   "metadata": {},
   "source": [
    "### For NR5A2:\n",
    "\n",
    "__Homer__\n",
    "1. get start\n",
    "```shell\n",
    "# for input\n",
    "makeTagDirectory ~/CSE185FinalProject/Figure_4e\\&f/data/tagdirs/input ~/CSE185FinalProject/Figure_4e\\&f/data/GSM846874_AlnInput.txt\n",
    "# for NR5a2\n",
    "makeTagDirectory ~/CSE185FinalProject/Figure_4e\\&f/data/tagdirs/NR5a2 ~/CSE185FinalProject/Figure_4e\\&f/data/GSM846875_AlnNR5a2ChIP.txt\n",
    "```\n",
    "2. find peak\n",
    "```shell\n",
    "findPeaks ~/CSE185FinalProject/Figure_4e\\&f/data/tagdirs/NR5a2 -i ~/CSE185FinalProject/Figure_4e\\&f/data/tagdirs/input -style factor -o auto\n",
    "```\n",
    "3. convert to bed \n",
    "```shell\n",
    "pos2bed.pl ~/CSE185FinalProject/Figure_4e\\&f/data/tagdirs/NR5a2/peaks.txt > ~/CSE185FinalProject/Figure_4e\\&f/data/NR5a2.peaks.bed\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f75386e6",
   "metadata": {},
   "source": [
    "#### Motif data location\n",
    "- AP-1: `~/CSE185FinalProject/Figure_4e\\&f/data/GSM537983_Sample1.ThioMac-PU.1.bed`\n",
    "- NR5a2: `~/CSE185FinalProject/Figure_4e\\&f/data/NR5a2.peaks.bed`"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1cc0dcfd",
   "metadata": {},
   "source": [
    "### Intersection with integrative location\n",
    "```shell\n",
    "# for AP-1\n",
    "intersectBed -wa \\\n",
    "               -a ~/CSE185FinalProject/Figure_4e\\&f/data/integrative.bed \\\n",
    "               -b ~/CSE185FinalProject/Figure_4e\\&f/data/GSM537983_Sample1.ThioMac-PU.1.bed \\\n",
    "> ~/CSE185FinalProject/Figure_4e\\&f/data/integrative_AP-1.bed\n",
    "\n",
    "# for NR5a2\n",
    "intersectBed -wa \\\n",
    "               -a ~/CSE185FinalProject/Figure_4e\\&f/data/integrative.bed \\\n",
    "               -b ~/CSE185FinalProject/Figure_4e\\&f/data/NR5a2.peaks.bed \\\n",
    "> ~/CSE185FinalProject/Figure_4e\\&f/data/integrative_NR5a2.bed\n",
    "```\n",
    "\n",
    "### lift over to mm10\n",
    "https://genome.ucsc.edu/cgi-bin/hgLiftOver\n",
    "\n",
    "__lifted file location__\n",
    "- AP-1: `~/CSE185FinalProject/Figure_4e\\&f/data/hglft_genome_integrative_AP-1.bed`\n",
    "- NR5a2: `~/CSE185FinalProject/Figure_4e\\&f/data/hglft_genome_integrative_NR5a2.bed`\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cbf27f02",
   "metadata": {},
   "source": [
    "### Intersection with integrative location\n",
    "```shell\n",
    "# for AP-1\n",
    "intersectBed -wa \\\n",
    "               -a ~/CSE185FinalProject/Figure_4e\\&f/data/scATAC.bed \\\n",
    "               -b ~/CSE185FinalProject/Figure_4e\\&f/data/hglft_genome_integrative_AP-1.bed \\\n",
    "               -f 1.0 \\\n",
    "> ~/CSE185FinalProject/Figure_4e\\&f/data/scATAC_AP-1.bed\n",
    "\n",
    "# for NR5a2\n",
    "intersectBed -wa \\\n",
    "               -a ~/CSE185FinalProject/Figure_4e\\&f/data/scATAC.bed \\\n",
    "               -b ~/CSE185FinalProject/Figure_4e\\&f/data/hglft_genome_integrative_NR5a2.bed \\\n",
    "               -f 1.0 \\\n",
    "> ~/CSE185FinalProject/Figure_4e\\&f/data/scATAC_NR5a2.bed\n",
    "```"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
